### PR TITLE
Makes the lib compatible with Node 4.1.1

### PR DESCRIPTION
**Pull request by hoffmannjan**
Tested this on project that I'm working now and everything seems to be OK 👍

### CODE REVIEW

**Comment by jasongin** [comment]
Thanks, and sorry for the delay in getting to this.

I'm not excited about taking on the added step of babel compilation, just to support Node 4. Maybe I could be convinced though. I understand a substantial number of people still run Node 4.

**Comment by jasongin** [comment]
I think a better solution might be to just change the source code to be Node 4 compatible. I haven't tried, but I think it shouldn't be too hard:

 1. Change all use of `let` and `const` to `var`.
 2. Replace some light use of ES6 `Promise` with either a promise library, or just change to callbacks.

**Comment by hoffmannjan** [comment]
Sure, I can do this :).
I was only thinking that It might be kind of pain for you to develop this lib in older ES version.

**Comment by jasongin** [comment]
If this is done, then I can update https://github.com/sandeepmistry/noble/pull/646 to use this package with Node 4 also.

**Comment by jrobeson** [comment]
now that node 4.x is EOL is this still worth doing?

**Comment by jrobeson** [comment]
I'm pushing to make 6.x the minimum version for the next version of noble btw.

File: src/bindings.js [hunk 1]
```diff
@@ -514,7 +514,7 @@ NobleBindings.prototype._onAdvertisementWatcherReceived = function(sender, e) {
 		var manufacturerData = manufacturerSections[0];
 		deviceRecord.manufacturerData = rt.toBuffer(manufacturerData.data);
 		let companyIdHex = manufacturerData.companyId.toString(16);
-		let toAppend = Buffer.allocUnsafe(2);
+		let toAppend = new Buffer(2);
```
Comment (jrobeson) [inline src/bindings.js]: this will spew warning errors in node 10.x

**Comment by jasongin** [comment]
> now that node 4.x is EOL is this still worth doing?

I think not, therefore I'm going to go ahead and close this PR. If anyone objects, please speak up!

**Pull request closed without merging**